### PR TITLE
Add a notification portal backend

### PR DESCRIPTION
**Pull request by matthiasclasen**
This just forwards the calls to gnome-shell.

### CODE REVIEW

File: Makefile.am [hunk 1]
```diff
@@ -64,6 +69,8 @@ xdg_desktop_portal_gtk_SOURCES = \
 	filechooser.c			\
 	appchooser.h			\
 	appchooser.c			\
+	notification.h			\
+	notification.c			\
```
Comment (alexlarsson) [inline Makefile.am]: Forgot to add these.

**Comment by alexlarsson** [comment]
These APIs have "gtk" in the name. I know there has been some historic tension about the notification APIs, but I'm unaware of the details. Do we also want to support some other kind of notification?

**Comment by matthiasclasen** [comment]
I would prefer if we could avoid this altogether and just use the gio notification support on the portal side, like we do for GProxyResolver and GNetworkMonitor. But sadly, the gio notification is not exposed in a suitable way for that; it is part of GApplication. That would give use freedesktop notification support for free.

The in-the-sandbox counterpart for this lives in the portal branch of glib, btw.

**Comment by matthiasclasen** [comment]
I've added support for org.freedesktop.Notifications now, largely by copying code from the GIO notification backend. Seems to work - there's a simple test in portal-test now.

**Pull request merged**